Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/dialogs/IniEditorDialog.java

```diff
@@ -23,125 +26,133 @@
-                sb.append(targets[i]);
-                sb.append(System.lineSeparator());
-                count += targets[i].length() + 2;
-                cursorPos = count;
-            }
-            else if(indices[i] > cursorPos)
-            {
-                cursorPos = indices[i] + targets[i].length() + 1;
-            }
+          int index = line.indexOf(targets[i]);
+          if(index != -1)
+          {
+            indices[i] = count + index;
+          }
         }
+        //
+        count += line.length() + 1;
+        sb.append(line);
+        sb.append(System.lineSeparator());
+        line = reader.readLine();
+      }
+    }
+    catch (Exception e)
+    {
+      Log.error(e.getMessage());
+      onLoadError();
+    }
 
-        String content = sb.toString();
-        if(cursorPos > content.length())
-            cursorPos = content.length();
-        editCode.setText(content);
-        editCode.setSelection(cursorPos);
+    int cursorPos = 0;
+    for(int i = 0; i < targets.length; ++i)
+    {
+      if(indices[i] < 0)
+      {
+        sb.append(System.lineSeparator());
+        sb.append(targets[i]);
+        sb.append(System.lineSeparator());
+        count += targets[i].length() + 2;
+        cursorPos = count;
+      }
+      else if(indices[i] > cursorPos)
+      {
+        cursorPos = indices[i] + targets[i].length() + 1;
+      }
     }
 
-    private void SaveIniContent(String gameId, String content)
+    String content = sb.toString();
+    if(cursorPos > content.length())
+      cursorPos = content.length();
+    editCode.setText(content);
+    editCode.setSelection(cursorPos);
+  }
+
+  private void saveIniContent(String gameId, String content)
+  {
+    String filename = DirectoryInitialization.getLocalSettingFile(gameId);
+    boolean saved = false;
+    try
     {
-        String filename = DirectoryInitialization.getLocalSettingFile(gameId);
-        //String[] lines = content.split(System.lineSeparator());
-        boolean saved = false;
-        try
-        {
-            BufferedWriter writer = new BufferedWriter(new FileWriter(filename));
-            writer.write(content);
-            writer.close();
-            saved = true;
-        }
-        catch (Exception e)
-        {
-            // ignore
-        }
-        Toast.makeText(getContext(), saved ? R.string.settings_saved : R.string.error,
-                Toast.LENGTH_SHORT).show();
+      BufferedWriter writer = new BufferedWriter(new FileWriter(filename));
+      writer.write(content);
+      writer.close();
+      saved = true;
     }
-}
+    catch (Exception e)
+    {
+      Log.error(e.getMessage());
+      onLoadError();
+    }
+
+    Toast.makeText(getContext(), saved ? R.string.settings_saved : R.string.error,
+      Toast.LENGTH_SHORT).show();
+  }
 
+  public void onLoadError()
+  {
+    TextView textError = mViewGroup.findViewById(R.id.ini_editor_error);
+    textError.setVisibility(View.GONE);
```

Comment:
Shouldn't this be `View.VISIBLE`?

Reply:
If I do that the text appears on the XML anyway

Reply:
Sorry, I don't understand. Could you clarify?

Reply:
If I make it visible, the text appears in the xml even without failing.

Reply:
So you mean... If you use `View.VISIBLE` here, then the user sees the text when they run Dolphin, even if no error occurred?

If that's the case, I think either you're calling this method in a situation where you shouldn't call it, or there is actually an error.